File: Week 2 - Arrays/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    //Ask user for text input
    string text = get_string("Text: ");
    

    int num_letters = 0;
    int num_words = 0;
    int num_sentences = 0;
    
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        //Calculate number of words by counting the number of single spaces between characters
        if (isspace(text[i]))
        {
            num_words++;
        }
        
        //Calculate number of sentences by counting punctation marks
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            num_sentences++;
        }
        
        //Convert character to lowercase, and increase number of letters by 1 if character is a letter
        int letter = tolower(text[i]);
        if (letter >= 97 && letter <= 122)
        {
            num_letters++;
        }
    }
    
    //Add the final word to your word count because the sentence will end with punctation rather than a space
    if (num_words > 0)
    {
        num_words++;
    }
    
    //Calculate the average number of letters and sentences per 100 words to use in  Coleman-Liau index
    double l = (num_letters * (100.00 / num_words));
    double s = (num_sentences * (100.00 / num_words));
    double index = 0.0588 * (l) - 0.296 * (s) - 15.8;
    
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}

File: Week 6 - Python/pset6/sentimental-readability/readability.py


def main():
    text = input("Text: ")

    num_letters = letter_count(text)

    # Split string into list of words
    words_list = text.split(' ')

    num_words = word_count(words_list)
    num_sentences = sentence_count(words_list)

    # Initialize variables for formula
    L = (num_letters/num_words) * 100
    S = (num_sentences/num_words) * 100
    grade = round(0.0588 * L - 0.296 * S - 15.8)
    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f'Grade {round(grade)}')


# Count all letters within text input
def letter_count(text):
    count = 0
    for char in text:
        if char.isalpha():
            count += 1
    return count

# Returns the word count via the length of the input list


def word_count(words_list):
    return len(words_list)

# Returns the number of sentences by checking the last digit of every word


def sentence_count(words_list):
    count = 0
    stop = ('...', '.', '?', '!', '!!!')
    for word in words_list:
        if word[-1] in stop:
            count += 1
    return count


if __name__ == "__main__":
    main()